platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java

```diff
@@ -63,6 +64,29 @@
     Project getProject(String workspace, String projectPath) throws ForbiddenException, ServerException, NotFoundException;
 
     /**
+     * Configure project after import. If project wasn't found in workspace it will be configured as blank.
+     *
+     * @param storage
+     *         storage from which project was imported
+     * @param workspaceId
+     *         workspace id
+     * @throws ServerException
+     *         if other error occurs
+     * @throws IOException
+     * @throws ForbiddenException
+     *         if user which perform operation doesn't have access to the requested project
+     * @throws NotFoundException
+     *         when project not found
+     * @throws ConflictException
+     *         if operation causes conflict
+     */
+    void configureProjectAfterImport(SourceStorageDto storage, String workspaceId) throws ServerException,
```

Comment:
I don't see the contract for `null` values, you should define this contract see [example](https://github.com/codenvy/che-core/blob/4.0/platform-api/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/HttpJsonRequest.java#L119-L122)